corehq/apps/consumer_user/signals.py

```diff
@@ -0,0 +1,80 @@
+from dimagi.utils.logging import notify_exception
+from dimagi.utils.web import get_site_domain
+from django.db.utils import IntegrityError
+from django.template.loader import render_to_string
+from django.utils.encoding import force_bytes
+from django.urls import reverse
+from corehq.apps.consumer_user.models import ConsumerUserInvitation
+from corehq.apps.consumer_user.models import ConsumerUserCaseRelationship
+from corehq.form_processor.models import CommCareCaseSQL
+from corehq.form_processor.signals import sql_case_post_save
+from corehq.apps.hqwebapp.tasks import send_html_email_async
+from django.utils.http import urlsafe_base64_encode
+from django.core.signing import TimestampSigner
+from corehq.apps.hqcase.utils import update_case
+
+
+def send_email_case_changed_receiver(sender, case, **kwargs):
+    try:
+        if case.type != 'commcare-caseuserinvitation':
+            return
+        try:
+            email = case.get_case_property('email')
+        except IntegrityError:
+            email = 'challarao@beehyv.com'
+        try:
+            updated_by = case.get_case_property('updated_by')
+        except IntegrityError:
+            updated_by = None
+        try:
+            invitation = ConsumerUserInvitation.objects.filter(case_id=case.case_id,
+                                                               domain=case.domain,
+                                                               active=True).latest('invited_on')
+            if invitation:
+                if updated_by and updated_by == 'patient' and email == invitation.email and not case.closed:
+                    return
+                invitation.active = False
+                ConsumerUserCaseRelationship.objects.filter(case_id=case.case_id,
```

Comment:
We have a unique_together constraint on case_id, domain in ConsumerUserCaseRelationship Model. In case if there are any rows with same case_id and domain(previous invitation might have been accepted right) so we are deleting it. The change is still there. If my understanding is wrong, let me know I will delete that code